src/components/SingleArticle.jsx

```diff
@@ -25,6 +27,19 @@ const SingleArticle = () => {
     return <Loading dynamicText={'article'} />;
   }
 
+  const upVote = event => {
+    patchArticleVotesBy(article_id, +1);
+    setArticle(currArticle => {
+      return { ...currArticle, votes: currArticle.votes + 1 };
+    });
+  };
+  const downVote = () => {
+    patchArticleVotesBy(article_id, -1);
+    setArticle(currArticle => {
+      return { ...currArticle, votes: currArticle.votes - 1 };
+    });
+  };
+
```

Comment:
Nice implementation of optimistic rendering, though you're missing some error handling if the API request to update the vote doesn't go through, here are some notes to help
https://notes.northcoders.com/courses/js-front-end/ui-behaviours-and-optimistic-rendering#error-handling
If the API request to change the vote doesn't work (like if the internet suddenly drops or something) we need to catch that error and update the state to let the user know that vote hasn't gone through